composeApp/src/commonTest/kotlin/com/arkivanov/minesweeper/game/DefaultGameComponentTest.kt

```diff
@@ -31,10 +31,46 @@ class DefaultGameComponentTest {
     }
 
     @Test
-    fun WHEN_created_THEN_stopwatch_on_START() {
-        assertEquals(GameStatus.INITIALIZED, state.gameStatus)
+    fun GIVEN_created_WHEN_cell_clicked_THEN_status_STARTED() {
+        clickCellPrimary(x = 0, y = 0)
+
+        assertEquals(GameStatus.STARTED, state.gameStatus)
+    }
+
+    @Test
+    fun GIVEN_created_WHEN_cell_clicked_THEN_timer_shows_0() {
+        clickCellPrimary(x = 0, y = 0)
+
         assertEquals(0, state.timer)
     }
 
-    // TODO: Write more tests (need a bit dive in to work with TestCoroutineScheduler
-}
\ No newline at end of file
+    @Test
+    fun GIVEN_game_started_WHEN_one_second_passed_THEN_timer_shows_1() {
+        clickCellPrimary(x = 0, y = 0)
+
+        coroutineScheduler.advanceTimeBy(1.seconds)
+        coroutineScheduler.runCurrent()
+
+        assertEquals(1, state.timer)
+    }
+
+    @Test
+    fun GIVEN_game_started_WHEN_click_on_mine_cell_THEN_timer_STOPS() {
+        clickCellPrimary(x = 0, y = 0)
+
+        coroutineScheduler.advanceTimeBy(10.seconds)
+        coroutineScheduler.runCurrent()
+
+        clickCellPrimary(x = 10, y = 10)
```

Comment:
I think this cell is not necessarily a mine, as the test name says? It would be nice to test both cases: when we click on a mine (fail) and when we win the game. In this case, I think we can create the component with default settings.

We can change our `clickCellPrimary` function to the following:

```kotlin
    private fun clickCellPrimary(location: Location) {
        gameComponent.onCellTouchedPrimary(x = location.x, y = location.y)
        gameComponent.onCellReleased(x = location.x, y = location.y)
    }
```

So it can be called manually like `clickCellPrimary(0 by 0)`, and also we can find/iterate cells from the state and click.

For example, we can click on a mine as follows:

```kotlin
        val (mineLocation) = state.grid.entries.first { (_, cell) -> cell.value.isMine }
        clickCellPrimary(mineLocation)
```

And we can click on all non-mine cells as follows:

```kotlin
        state.grid.filterValues { it.status.isClosed && !it.value.isMine }.forEach { (location, _) ->
            clickCellPrimary(location)
        }
```